pkg/cmd/infra/router/template.go

```diff
@@ -287,6 +293,77 @@ func parseCaptureHeaders(in string) ([]templateplugin.CaptureHTTPHeader, error)
 	return captureHeaders, nil
 }
 
+func sanitizeHeaderValue(headerValue string) string {
+	sanitizeHeaderValue := strings.ReplaceAll(headerValue, "'", `'\''`)
+	return "'" + sanitizeHeaderValue + "'"
+}
+
+func parseHeadersToBeSetOrDeleted(in string) ([]templateplugin.HTTPHeader, error) {
+	var captureHeaders []templateplugin.HTTPHeader
+	var capture templateplugin.HTTPHeader
+	var err error
+	if len(in) == 0 {
+		return captureHeaders, fmt.Errorf("encoded header string not present.")
+	}
+	if len(in) > 0 {
+		for _, header := range strings.Split(in, ",") {
+			parts := strings.Split(header, ":")
+			if len(parts) < 2 {
```

Comment:
There doesn't seem to be any handling for if `len(parts) > 3`. It looks like the entire malformed item will get silently dropped and the list of header actions will continue to be processed. That wouldn't cause a crash or anything severe, but if `len(parts) < 2` causes an error to be returned, I'd expect `len(parts) > 3` to also return an error.

This looks like a reasonable place to use `switch`/`case` instead of a series of `if` statements, and it would also be easy to get the behavior I'm suggesting. For example:
```go
switch len(parts) {
case 3:
    // parse header:value:action
case 2:
    // parse header:action
default:
    // return error
}
```

Reply:
done

Reply:
@rfredette nice suggestion. thanks!